### PR TITLE
Update Glass Mapper to v5

### PR DESCRIPTION
**Pull request by yuya-yuliya**
- update Glass Mapper to v5
- adjust code generation templates

### CODE REVIEW

**Review by ghost — COMMENTED** [review]
Current request and branch will be closed. In the next request please:
1. Leave only items related to upgrade to v5 (package.config, .csproj, etc.)
2. Please, **no side refactoring**
3. Don't move config files to Glass folder, it breaks site after deploy
Close current request and delete branch after creating a new one.

File: src/Foundation/GlassMapper/code/App_Start/GlassMapperSc.cs [hunk 1]
```diff
@@ -20,60 +20,60 @@
...
-    using Glass.Mapper;
-    using Glass.Mapper.Maps;
-    using Glass.Mapper.Sc.Configuration.Fluent;
-    using Glass.Mapper.Sc.IoC;
-    using Sitecore.Pipelines;
-
-    public class GlassMapperSc
-    {
-        public void Process(PipelineArgs args)
+	public class GlassMapperSc : Glass.Mapper.Sc.Pipelines.Initialize.GlassMapperSc
```
[the remark this answers is not in the document]
Comment (yuya-yuliya) [reply src/Foundation/GlassMapper/code/App_Start/GlassMapperSc.cs]: This is Glass Mapper auto generated file

File: src/Foundation/GlassMapper/code/App_Start/GlassMapperSc.cs [hunk 2]
```diff
@@ -20,60 +20,60 @@
...
+	{
+        public override IDependencyResolver CreateResolver()
         {
-            Start();
```
[the remark this answers is not in the document]
Comment (yuya-yuliya) [reply src/Foundation/GlassMapper/code/App_Start/GlassMapperSc.cs]: This is Glass Mapper auto generated file

File: src/Foundation/GlassMapper/code/App_Start/GlassMapperSc.cs [hunk 3]
```diff
@@ -20,60 +20,60 @@
...
-
-            GlassMapperScCustom.PostLoad(resolver);
-
-
-            //EditFrameBuilder.EditFrameItemPrefix = "Glass-";
+          return loaders1.Concat(loaders2).ToArray();
         }
 
-        public static void LoadConfigurationMaps(IDependencyResolver resolver, Context context)
+        public override void LoadConfigurationMaps(IDependencyResolver resolver, Glass.Mapper.Context context)
```
[the remark this answers is not in the document]
Comment (yuya-yuliya) [reply src/Foundation/GlassMapper/code/App_Start/GlassMapperSc.cs]: This is Glass Mapper auto generated file

File: src/Foundation/GlassMapper/code/App_Start/GlassMapperSc.cs [hunk 4]
```diff
@@ -20,60 +20,60 @@
...
         {
             var dependencyResolver = resolver as DependencyResolver;
-            if (dependencyResolver == null) return;
+            if (dependencyResolver == null)
```
[the remark this answers is not in the document]
Comment (yuya-yuliya) [reply src/Foundation/GlassMapper/code/App_Start/GlassMapperSc.cs]: This is Glass Mapper auto generated file

File: src/Foundation/GlassMapper/code/App_Start/GlassMapperSc.cs [hunk 5]
```diff
@@ -20,60 +20,60 @@
...
+            {
+                return;
+            }
 
             if (dependencyResolver.ConfigurationMapFactory is ConfigurationMapConfigFactory)
+            {
```
[the remark this answers is not in the document]
Comment (yuya-yuliya) [reply src/Foundation/GlassMapper/code/App_Start/GlassMapperSc.cs]: This is Glass Mapper auto generated file

File: src/Foundation/GlassMapper/code/App_Start/GlassMapperSc.cs [hunk 6]
```diff
@@ -20,60 +20,60 @@
...
                 GlassMapperScCustom.AddMaps(dependencyResolver.ConfigurationMapFactory);
+            }
 
             IConfigurationMap configurationMap = new ConfigurationMap(dependencyResolver);
-            var configurationLoader = configurationMap.GetConfigurationLoader<SitecoreFluentConfigurationLoader>();
+            SitecoreFluentConfigurationLoader configurationLoader = configurationMap.GetConfigurationLoader<SitecoreFluentConfigurationLoader>();
```
[the remark this answers is not in the document]
Comment (yuya-yuliya) [reply src/Foundation/GlassMapper/code/App_Start/GlassMapperSc.cs]: This is Glass Mapper auto generated file

File: src/Foundation/GlassMapper/code/App_Start/GlassMapperSc.cs [hunk 7]
```diff
@@ -20,60 +20,60 @@
...
             context.Load(configurationLoader);
+
+            base.LoadConfigurationMaps(resolver, context);
         }
-    }
-}
 
+	    public override void PostLoad(IDependencyResolver dependencyResolver)
+	    {
+			GlassMapperScCustom.PostLoad();
```
[the remark this answers is not in the document]
Comment (yuya-yuliya) [reply src/Foundation/GlassMapper/code/App_Start/GlassMapperSc.cs]: This is Glass Mapper auto generated file

File: src/Foundation/GlassMapper/code/App_Start/GlassMapperScCustom.cs [hunk 8]
```diff
@@ -13,45 +13,37 @@
...
+    public static  class GlassMapperScCustom
     {
-        public static IDependencyResolver CreateResolver()
-        {
-            var config = new Config();
-
-            var dependencyResolver = new DependencyResolver(config);
-            // add any changes to the standard resolver here
+		public static IDependencyResolver CreateResolver(){
+			var config = new Glass.Mapper.Sc.Config();
```
[the remark this answers is not in the document]
Comment (yuya-yuliya) [reply src/Foundation/GlassMapper/code/App_Start/GlassMapperScCustom.cs]: This is Glass Mapper auto generated file

File: src/Foundation/GlassMapper/code/App_Start/GlassMapperScCustom.cs [hunk 9]
```diff
@@ -13,45 +13,37 @@
...
 
-            dependencyResolver.Finalise();
```
[the remark this answers is not in the document]
Comment (yuya-yuliya) [reply src/Foundation/GlassMapper/code/App_Start/GlassMapperScCustom.cs]: This is Glass Mapper auto generated file. Lines removed Glass Mapper authors

File: src/Foundation/GlassMapper/code/App_Config/Include/Glass.Mapper.Sc.config [hunk 10]
```diff
@@ -1,26 +0,0 @@
-<configuration xmlns:patch="http://www.sitecore.net/xmlconfig/">
```
[the remark this answers is not in the document]
Comment (yuya-yuliya) [reply src/Foundation/GlassMapper/code/App_Config/Include/Glass.Mapper.Sc.config]: Ok. I'll move new configs to Include folder. Thank you

**Pull request merged**